### ISSUE TITLE
Inherent associated type doesn't work.

### ISSUE THREAD

**Issue by crlf0710** [rust-lang/rust#85204]
I tried this code:

```rust
#![feature(inherent_associated_types)]
struct S;
impl S {
    type Foo = Vec<i32>;
}

fn main() {
    let s = S::Foo::default();
}
```

I expected to see this happen: It compiles.

Instead, this happened: It fails with error:
```text
error[E0223]: ambiguous associated type
 --> src/main.rs:8:13
  |
8 |     let s = S::Foo::default();
  |             ^^^^^^ help: use fully-qualified syntax: `<S as Trait>::Foo`

error: aborting due to previous error; 1 warning emitted
```

**Comment by FabianWolff** [comment]
I think this is expected, because inherent associated types are not implemented yet (see the tracking issue #8995), only the feature gate and declaring (but not using) them (#82516).

**Comment by ssbr** [comment]
The same example also won't work for non-inherent associated types: https://github.com/rust-lang/rust/issues/38078#issuecomment-887044580

**Comment by jhpratt** [comment]
This is definitely expected behavior and not a bug, as inherent associated types are not implemented. In my opinion this should be closed in favor of the tracking issue (#8995).

@rustbot label -C-bug
